### PR TITLE
Implement highlighting on hover

### PR DESCRIPTION
**Pull request by KaroMourad**
Fix: 
 - style: clipPath sizes for circles
 - bug: useResizeObserver hook
 - optimize: draw-attributes function speed on mouse-over
 
Add: 
 - Ability to select highlight modes from controls sidebar
 - Styles for highlighted lines
 - Implement 'selectors' functionality for multi-highlighted lines
 - Implement mouse-leave functionality

### CODE REVIEW

**Review by roubkar — APPROVED** [review]
lgtm

File: aim/web/ui_v2/src/utils/d3/drawHoverAttributes.ts [hunk 1]
```diff
@@ -68,85 +63,133 @@ function drawHoverAttributes(props: IDrawHoverAttributesProps): void {
...
+        .classed('active', false);
+
+      attributesNodeRef.current
+        .select(`[id=Circle-${closestCircleRef.current.key}]`)
+        .attr('r', CircleEnum.Radius)
+        .classed('active', false);
+
+      attributesNodeRef.current.select('.HoverLine-y').remove();
+
+      yAxisLabelNodeRef.current?.remove();
```
Comment (gorarakelyan) [inline aim/web/ui_v2/src/utils/d3/drawHoverAttributes.ts:191]: I guess here we should reset opacity of not highlight lines as well

File: aim/web/ui_v2/src/services/api/metrics/metricsService.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ function getMetricsData() {
     call: () => ({
       then: (resolve: (data: IRun[]) => void, reject?: unknown) => {
         setTimeout(() => {
-          const mock = generateMetrics(500, 200);
+          const mock = generateMetrics(100, 50);
```
Comment (gorarakelyan) [inline aim/web/ui_v2/src/services/api/metrics/metricsService.ts:15]: It seems interactions slow down when the number of visualized lines becomes greater than 200.

Comment (roubkar) [reply aim/web/ui_v2/src/services/api/metrics/metricsService.ts:15]: Yeah, indeed, but this is actually not related to this PR only.
We've noticed the same issue for hover attributes rendering on screens with high resolution.
We will make an investigation and try to resolve this issue within a separate PR.

Comment (gorarakelyan) [reply aim/web/ui_v2/src/services/api/metrics/metricsService.ts:15]: Got it, great.

File: aim/web/ui_v2/src/utils/d3/drawHoverAttributes.ts (hunk 1 above)
Comment (KaroMourad) [reply aim/web/ui_v2/src/utils/d3/drawHoverAttributes.ts:191]: You mean on mouse leave we should reset highlight mode to highlight-off mode ?

Comment (roubkar) [reply aim/web/ui_v2/src/utils/d3/drawHoverAttributes.ts:191]: No, we just need to reset the highlighted line state, so no line differs from the others.

Comment (KaroMourad) [reply aim/web/ui_v2/src/utils/d3/drawHoverAttributes.ts:191]: ok, i will fix this

File: aim/web/ui_v2/src/types/components/LineChart/LineChart.d.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { CurveEnum } from 'utils/d3';
+import HighlightEnum from '../../../components/HighlightModesPopover/HighlightEnum';
```
Comment (arsengit) [inline aim/web/ui_v2/src/types/components/LineChart/LineChart.d.ts:2]: Can we import modules with absolute path to avoid long names ?

**Review by arsengit — APPROVED** [review]

File: aim/web/ui_v2/src/types/components/LineChart/LineChart.d.ts (hunk 3 above)
Comment (KaroMourad) [reply aim/web/ui_v2/src/types/components/LineChart/LineChart.d.ts:2]: of course , it's automated imports but i will be more attentive about that , thanks

**Review by gorarakelyan — APPROVED** [review]
looks great

**Pull request merged**